contracts/factories/DelegatedManagerFactory.sol

```diff
@@ -0,0 +1,378 @@
+     *
+     * (Note: This flow should work well for SetTokens managed by an EOA. However, existing
+     * contract-managed Sets may need to have their ownership temporarily transferred to an EOA when
+     * migrating. We don't anticipate high demand for this migration case though.)
+     *
+     * @param  _setToken         Instance of SetToken to migrate to the DelegatedManager system
+     * @param  _owner            Address to set as the DelegateManager's `owner` role
+     * @param  _methodologist    Address to set as the DelegateManager's methodologist role
+     * @param  _operators        List of operators authorized for the DelegateManager
+     * @param  _assets           List of assets DelegateManager can trade. When empty, manager can trade any asset
+     * @param  _extensions       List of extensions authorized for the DelegateManager
+     *
+     * @return (address) Address of the created DelegatedManager
+     */
+    function createManager(
+        ISetToken _setToken,
+        address _owner,
+        address _methodologist,
+        address[] memory _operators,
+        address[] memory _assets,
+        address[] memory _extensions
+    )
+        external
+        returns (address)
+    {
+        if (_assets.length != 0) {
+            _validateComponentsIncludedInAssetsList(_setToken.getComponents(), _assets);
+        }
+
+        require(msg.sender == _setToken.manager(), "Must be manager");
+
+        DelegatedManager manager = _deployManager(
+            _setToken,
+            _methodologist,
+            _extensions,
+            _operators,
+            _assets
+        );
+
+        _setInitializationState(_setToken, address(manager), _owner);
+
+        return address(manager);
+    }
+
+    /**
+     * ONLY DEPLOYER: Wires SetToken, DelegatedManager, global manager extensions, and modules together
+     * into a functioning package.
+     *
+     * @param  _setToken                Instance of the SetToken
+     * @param  _ownerFeeSplit           Percent of fees in precise units (10^16 = 1%) sent to operator, rest to methodologist
+     * @param  _ownerFeeRecipient       Address which receives operator's share of fees when they're distributed
+     * @param  _initializeTargets       List of addresses of any extensions or modules which need to be initialized
+     * @param  _initializeBytecode      List of bytecode encoded calls to relevant target's initialize function
+     */
+    function initialize(
+        ISetToken _setToken,
+        uint256 _ownerFeeSplit,
+        address _ownerFeeRecipient,
+        address[] memory _initializeTargets,
+        bytes[] memory _initializeBytecode
+    )
+        external
+    {
+        require(initializeState[_setToken].isPending, "Manager must be awaiting initialization");
+        require(msg.sender == initializeState[_setToken].deployer, "Only deployer can initialize manager");
+        _initializeTargets.validatePairsWithArray(_initializeBytecode);
```

Comment:
Good flag. 

I can't think of a scenario where we would call `initialize` on a module or extension twice in the same `factory.initialize` call - they should be "one and done". 

I introduced a bug in my PR by removing the requirement that we have at least one extension. This method requires the targets array be non-empty.

Reply:
Restored required that there is at least one item in the extensions array when calling either create method in [2f990ea](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/2f990eabd47987d80d4e0d6a8c93b17a1ec483fd)

Reply:
yeah it also makes it sort of a soft requirement that you can't initialize and do something else with the module during the initialization process. Let's actually note that so that auditors are aware. Due to uniqueness requirement no other actions with a module or extension can be taken in `initialize`. Good for them to know that potential attack surface is minimized

Reply:
Added comment noting this